app/main/forms.py

```diff
@@ -282,6 +285,20 @@ class GovukIntegerField(GovukTextInputFieldMixin, IntegerField):
     pass
 
 
+class GovukDecimalField(GovukTextInputFieldMixin, DecimalField):
+    pass
+
+
+class PostcodeSearchField(GovukTextInputFieldMixin, SearchField):
+    input_type = "Search"
+    param_extensions = {"classes": "govuk-input--width-10"}
+    validators = [
+        DataRequired(message="Enter a postcode."),
+        MustContainAlphanumericCharacters(),
+        Length(max=255, message="Service name must be 255 characters or fewer"),
```

Comment:
I think you left the message from somewhere else when it fails, also, can't we just check if it contains a valid postcode rather than just containing both letters and numbers since it's a very specific format.  Either with the giant proper postcode regex or a much simpler strip whitespace then check against something like`[a-z]{1,2}\d{1,2}[a-z]?\d[a-z]{2}`.  Having written that regex I'd probably just use a library for it like https://pypi.org/project/postcode-validator/.